subprojects/core/src/main/java/org/gradle/initialization/DefaultSettingsLoader.java

```diff
@@ -68,6 +71,24 @@ public SettingsInternal findAndLoadSettings(GradleInternal gradle) {
         return settings;
     }
 
+    private void deprecateWarningIfNecessary(StartParameter startParameter, SettingsInternal settings) {
+        if (startParameter.getSettingsFile() != null) {
+            return;
+        }
+
+        File projectDir = startParameter.getProjectDir() == null ? startParameter.getCurrentDir() : startParameter.getProjectDir();
+        if (settings.getSettingsDir().equals(projectDir)) {
+            // settings only project, see ProjectLoadingIntegrationTest.settingsFileGetsIgnoredWhenUsingSettingsOnlyDirectoryAsProjectDirectory
+            return;
+        }
+        for (ProjectDescriptor project : settings.getProjectRegistry().getAllProjects()) {
+            if (project.getProjectDir().equals(projectDir)) {
+                return;
+            }
+        }
+        DeprecationLogger.nagUserWith("Support for nested build without a 'settings.gradle' was deprecated and will be removed in Gradle 5.0. You should create a empty settings.gradle in " + projectDir.getAbsolutePath());
```

Comment:
:x: Since #3120, a _settings file_ could be named either `settings.gradle` or `settings.gradle.kts`. User messages should not display `settings.gradle` but `settings file` instead.

e.g.: https://github.com/gradle/gradle/blob/f8802b64016addda984cd7f0773b9d2f00bdffc1/subprojects/composite-builds/src/main/java/org/gradle/composite/internal/DefaultIncludedBuildFactory.java#L57

Reply:
Thanks for reminding!